### ISSUE TITLE
Analyze implementing interfaces

### ISSUE THREAD

**Issue by PaulWoitaschek** [afcastano/AutoValuePlugin#14]
Assuming I have an interface

``` java
public interface Interface {
   String value();
}
```

and an implementing model

``` java
@AutoValue public abstract class Model implements Interface {}
```

The plugin does not generate the `value()` method for the builder / constructor, unless I explicitly override it like

``` java
@AutoValue public abstract class Model implements Interface {
   @Override public abstract String value();
}
```

In case I don't explicitly override it, the fields are just missing like

``` java
public static Model create() {return new AutoValue_Model();}
```

**Comment by afcastano** [comment]
Good point, I'll see if I can implement this.

Thanks!

**Comment by vanniktech** [comment]
Has there been any progress regarding this?

**Comment by afcastano** [comment]
Hello!
This is implemented in version 1.0.0

I have submitted the file to jetBrains and it takes up to two days for them to approve it. Once approved, it should ask you to update.

I'll let you guys know anyway.

Sorry for the delay. I was too busy to work on this the last months, I hope this is still of any value for you.

Thanks.

**Comment by afcastano** [comment]
FYI, Jetbrains has approved the file. Now you can update to version 1.0.0 which fixes this issue. Let me know if it works so I can close this.

**Comment by radekkozak** [comment]
Hi, like other guys i've been looking for this as well and it seems to be working ok now with your new 1.0.0 version of the plugin - both for create method and for the Builder pattern. 

Thanks

**Comment by afcastano** [comment]
Cool, I will close this issue then!
Thanks!

**Comment by miquelbeltran** [comment]
Hello, since this PR was released I am getting weird results when I use the plugin with AutoValue classes that implement Parcellable or Map, because it adds the builder setters from the Parcellable or Map interfaces, which is something I don't want to do.

What could be done to avoid this?

**Comment by afcastano** [comment]
Yeah... that is an issue! Sorry I didn't catch this use case earlier.
I can definitely put together a quick patch to ignore methods from android.os.Parcelable, but looking for a more scalable solution, could you post an example of how would you use @AutoValue with classes that implement Map for example?

Thanks!

**Comment by PaulWoitaschek** [comment]
Maybe have a look at how AutoValue itself handles it.

**Comment by miquelbeltran** [comment]
No problem! I use Map with an auto-value-map library, to create Retrofit QueryMap parameters.

For example

```@AutoValue
public abstract class MyQueryMap implements Map<String, Object> {
    @Nullable
    @MapKey("query_parameter")
    public abstract Boolean getQueryParameter();
// ...
}
```

So when I generate the Builder methods it adds me these:

```
    @AutoValue.Builder
    public abstract static class Builder {

        public abstract Builder size(int size);

        public abstract Builder setEmpty(boolean newEmpty);

        public abstract Builder keySet(Set<K> keySet);

        public abstract Builder values(Collection<V> values);

        public abstract Builder entrySet(Set<Entry<K, V>> entrySet);

        public abstract Builder hashCode(int hashCode);
```

and the auto-value-map library I use: https://github.com/volders/auto-value-map

**Comment by afcastano** [comment]
Ok, this can potentially become a problem for every AutoValue extension that is implemented.
For example someone could come up with @AutoValue extension for lists... that would generate a `java.util.List` implementation with the property values as the list elements. It will be difficult to analyse every extension and try to come up with the things that I should generate or not.

One feasible solution I can imagine would be to provide a settings page for the plugin, in which the user can add the interfaces that he explicitly want to avoid when generating the methods.

It would be a one-off configuration for the whole project. So in your case @miquelbeltran you might need to add `android.os.Parcelable` and `java.util.Map` in the configuration.

This might take some time, so for the time being, I will just ignore `android.os.Parcelable` and `java.util.*` classes.

What do you guys think about this?

Thank you very much for your contribution!

**Comment by miquelbeltran** [comment]
I think it is a good solution.

I won't mind the configuration step. Although I would like to have the Parcelable and utils.* in there by default.

Thanks for your hard work!

**Comment by afcastano** [comment]
Ok, I have just released version 1.0.2 which ignores `android.os.Parcelable` and anything from `java.util.*`

You should be able to download the new version in a few hours.

The configuration page might take longer. I hope this fixes your problem. =)
